src/compiler/scala/tools/nsc/backend/jvm/BCodeHelpers.scala

```diff
@@ -35,15 +35,26 @@ abstract class BCodeHelpers extends BCodeIdiomatic with BytecodeWriters {
   /*
    * must-single-thread
    */
-  def getOutFolder(csym: Symbol, cName: String, cunit: CompilationUnit): _root_.scala.tools.nsc.io.AbstractFile = {
-    try {
-      outputDirectory(csym)
-    } catch {
-      case ex: Throwable =>
-        reporter.error(cunit.body.pos, s"Couldn't create file for class $cName\n${ex.getMessage}")
-        null
-    }
-  }
+  def getOutFolder(csym: Symbol, cName: String, cunit: CompilationUnit): _root_.scala.tools.nsc.io.AbstractFile = Option {
+      try {
+        outputDirectory(csym)
+      } catch {
+        case ex: Throwable =>
+          reporter.warning(cunit.body.pos, s"Couldn't find output folder for symbol source ${csym.name}. Dropping to compliation unit source.\n${ex.getMessage}")
```

Comment:
This translates to a compiler warning that the user can do nothing about. If anyone compiles with `-Xfatal-warnings` this will break their build, right?

I think you could log something if you think this is important, or better yet, find the root cause and fix it there. Can't the symbol itself carry whatever is missing compared to the "regular" case?